### PR TITLE
Weapon reload times

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

This makes it so that any weapon with a magazine takes time to reload. UNLESS you have the skillchip. I will be adding sources to get said skillchip later.

All guns take a standard 2 seconds to reload, however, you can change this by setting the reload_time. If set to 0, the gun instantly reloads.

Only applies to guns with magazines.

Only applies to tactical reloads e.g. putting a mag in when there's already one inserted.

## Why It's Good For The Game

People have been complaining about firefights not lasting long enough, this should rectify this. This is the first step towards it anyway.

## Changelog
:cl:
add: Weapon reload times + Skillchip
add: Alt+click weapon magazine removal.
/:cl:

### CODE REVIEW

**Comment by RatFromTheJungle** [comment]
does this apply to shotguns?

**Comment by Gandalf2k15** [comment]
> does this apply to shotguns?

No only guns with magazines

**Comment by Elli-Skala** [comment]
i have a bad feeling this will lead to the power creep problem, where rather sec or antags will get a direct buff from this (depending on who gets said planned skill chip) while normal crew members get even weaker in fights

**Comment by Gandalf2k15** [comment]
> i have a bad feeling this will lead to the power creep problem, where rather sec or antags will get a direct buff from this (depending on who gets said planned skill chip) while normal crew members get even weaker in fights

Only traitors will get it likely, or antag roles.

 Also power creep? Hello? This is literally the opposite as all weapons are getting a nerf

**Comment by MLGTASTICa** [comment]
> i have a bad feeling this will lead to the power creep problem, where rather sec or antags will get a direct buff from this (depending on who gets said planned skill chip) while normal crew members get even weaker in fights

normal crewmembers are not trained to handle military weapons and shouldn't act like they know how to tactically reload guns.

**Comment by Savotta** [comment]
wait so is sec just suddenly not trained in reloads or what 

(nvm heard that sec will able to reload in 0.5s, aka 75% less from feedback)

**Comment by gamermomentum** [comment]
Hey so I think this PR's kind of a terrible idea but uh if you really wanna do it 
Probably implement this as like a tiered skill or something that you can develop by drilling it. 

I used to work with a firearms instructor on some things and like uh. 
When he first handed me a new firearm, the reload was slow and clumsy. But before we moved onto the next one, I could reload it *very* quickly from basically any state of readiness. 

Basically let me level of my reload skill by drilling instead of only with skillchips because skillchips are dumb and cringe.

**Comment by Vizoplez** [comment]
> Hey so I think this PR's kind of a terrible idea but uh if you really wanna do it
> Probably implement this as like a tiered skill or something that you can develop by drilling it.
> 
> I used to work with a firearms instructor on some things and like uh.
> When he first handed me a new firearm, the reload was slow and clumsy. But before we moved onto the next one, I could reload it _very_ quickly from basically any state of readiness.
> 
> Basically let me level of my reload skill by drilling instead of only with skillchips because skillchips are dumb and cringe.

If you read through; its not skill-chip only, certain jobs get buffs or debuffs IIRC; all of sec have a 75% reduction last I checked in reload time, and antags can BUY a skillchip that allows them to speed-reload.

Nothing here prevents that - and its more a buff and a general nerf to antags and sec alike; along with guns overall.

**Comment by Gandalf2k15** [comment]
> Hey so I think this PR's kind of a terrible idea but uh if you really wanna do it
> Probably implement this as like a tiered skill or something that you can develop by drilling it.
> 
> I used to work with a firearms instructor on some things and like uh.
> When he first handed me a new firearm, the reload was slow and clumsy. But before we moved onto the next one, I could reload it _very_ quickly from basically any state of readiness.
> 
> Basically let me level of my reload skill by drilling instead of only with skillchips because skillchips are dumb and cringe.

This is a game and I'm not making it persistence based either because people do not have character mains. E.g. sec officer suddenly a cook.

**Comment by gamermomentum** [comment]
> 
> 
> > Hey so I think this PR's kind of a terrible idea but uh if you really wanna do it
> > Probably implement this as like a tiered skill or something that you can develop by drilling it.
> > I used to work with a firearms instructor on some things and like uh.
> > When he first handed me a new firearm, the reload was slow and clumsy. But before we moved onto the next one, I could reload it _very_ quickly from basically any state of readiness.
> > Basically let me level of my reload skill by drilling instead of only with skillchips because skillchips are dumb and cringe.
> 
> This is a game and I'm not making it persistence based either because people do not have character mains. E.g. sec officer suddenly a cook.

when I said drilling ig I meant in a round. 
If you wanna go practice reloading somewhere private, you stop being shit at reloading.

**Comment by Draganfrukts** [comment]
Make shotguns have a 0.2s delay and allow the shells to be stacked up to 6, then make it autofill

**Comment by ghost** [comment]
To what purpose does a slow-paced game like SS13 need for timed reloads?

 In a FPS shooter it usually only takes a single press of a button (sometimes it does it passively too), while in Space Station 13 has multiple clicks atleast 4-7 depending on hotkeys and that is disregarding mouse travel time (another thing virtually all FPS don't have). Instead of making weapons scarce, yet powerful, items that can make or break a combat encounter, this only makes it further a monkey's paw where if you're not [specified role here], then weapons are impractical and uneffective.
 
 I don't see the purpose in overcomplicating reloading for player when it would be better in my eyes to simply allow player choice and secondary factors to devalue Urist McAssistant from doing Security's job instead of directly affecting the mechanic itself.
 
c377e7756cbd90a4f5f430a92babd254

**Comment by Gandalf2k15** [comment]
> To what purpose does a slow-paced game like SS13 need for timed reloads?
> 
> In a FPS shooter it usually only takes a single press of a button (sometimes it does it passively too), while in Space Station 13 has multiple clicks atleast 4-7 depending on hotkeys and that is disregarding mouse travel time (another thing virtually all FPS don't have). Instead of making weapons scarce, yet powerful, items that can make or break a combat encounter, this only makes it further a monkey's paw where if you're not [specified role here], then weapons are impractical and uneffective.
> 
> I don't see the purpose in overcomplicating reloading for player when it would be better in my eyes to simply allow player choice and secondary factors to devalue Urist McAssistant from doing Security's job instead of directly affecting the mechanic itself.
> 
> c377e7756cbd90a4f5f430a92babd254

This only applies to tactical reloads

**Comment by ghost** [comment]
> This only applies to tactical reloads

My point still stands, this feels needless and overcomplicated.

**Comment by Gandalf2k15** [comment]
> > This only applies to tactical reloads
> 
> My point still stands, this feels needless and overcomplicated.

Not at all. Normies shouldn't be doing tac reloads. Good way make gunfights last longer

**Comment by Sethnov** [comment]
can we make it able to like buy the skill chip atleast ?

**Comment by SpookyTheFox** [comment]
The metric that makes firefights last longer is not action timers on reload after the fight is over.  The metric that makes fights last longer, is increasing the number of clicks required to horizontal the spaceman. The largest culprit of this, is too high damage per click from firearms. I am in favor of eliminating virtually all burst mode shooting, as it stacks on damage too quickly. Eliminating tactical reload entirely would help somewhat, but the total number of clicks to horizontal a spaceman should go up. The the most glaring problem is at the damage per bullet, and bullets per click.

Further, the primary users of magazine fed, high damage weaponry that instantly ends fights are security, why would they be exempt? That's neither realistic nor is it in the interest of game play. And having an optional skill chip for antags is a TC sink, leaves evidence in the form of the auto surgeon, and doesn't cover any other antag type that should rightfully have unfettered access. A trait that is automatically applied to all antags would be the correct solution, and help us get away from skillchips.

A better solution still though, would be to not mess with per-user reload times at all, and instead limit the damage per click from the plethora of guns that are two and three click fight enders.

**Comment by LeonY24** [comment]
got two questions
1. do you have to stand still? if yes it's terrible
2. is reloading proccess will be accompanied with reloading sound, like, mag out, mag in, click-clack?

**Comment by Gandalf2k15** [comment]
> The metric that makes firefights last longer is not action timers on reload after the fight is over. The metric that makes fights last longer, is increasing the number of clicks required to horizontal the spaceman. The largest culprit of this, is too high damage per click from firearms. I am in favor of eliminating virtually all burst mode shooting, as it stacks on damage too quickly. Eliminating tactical reload entirely would help somewhat, but the total number of clicks to horizontal a spaceman should go up. The the most glaring problem is at the damage per bullet, and bullets per click.
> 
> Further, the primary users of magazine fed, high damage weaponry that instantly ends fights are security, why would they be exempt? That's neither realistic nor is it in the interest of game play. And having an optional skill chip for antags is a TC sink, leaves evidence in the form of the auto surgeon, and doesn't cover any other antag type that should rightfully have unfettered access. A trait that is automatically applied to all antags would be the correct solution, and help us get away from skillchips.
> 
> A better solution still though, would be to not mess with per-user reload times at all, and instead limit the damage per click from the plethora of guns that are two and three click fight enders.

It only applies to tactical reloads. Normal reloading isn't affected.

**Comment by Gandalf2k15** [comment]
I've decided this isn't a good idea if I'm completely honest.
